server/service/src/sync/translations/master_list.rs

```diff
@@ -31,16 +71,49 @@ impl SyncTranslation for MasterListTranslation {
 
         let data = serde_json::from_str::<LegacyListMasterRow>(&sync_record.data)?;
 
-        let result = MasterListRow {
-            id: data.ID,
-            name: data.description,
-            code: data.code,
-            description: data.note,
+        let master_list = MasterListRow {
+            id: data.id.clone(),
+            name: data.description.clone(),
+            code: data.code.clone(),
+            description: data.note.clone(),
         };
 
-        Ok(Some(IntegrationRecords::from_upsert(
-            PullUpsertRecord::MasterList(result),
-        )))
+        let generate = generate_requisition_program(connection, data)?;
```

Comment:
I think below would be slightly cleaner (don't have to match none and create defaults that are not used
```rust
if let Some(generate) = generate_requisition_program(connection, data)? else {
  return Ok(Some(IntegrationRecords::from_upsert(
            PullUpsertRecord::MasterList(result),
        )))
}
```